backend/game/engine/player_queue.py

```diff
@@ -0,0 +1,45 @@
+import itertools
+
+
+class GamePlayers():
+
+    def __init__(self):
+        self.all_players = []
+        self.dead_players = []
+        self.alive_players = []
+        self.current_player = None
+
+    def _add_initial_player(self, player):
+        self.all_players.append(player)
+
+    def _remove_initial_player(self, player):
+        if player in self.all_players:
+            self.all_players.remove(player)
+
+    def set_player_order(self):
+        # TODO: set this based off of dice roll minigame
+        # Currently just the order in which they were added to the list
+        self.alive_players = self.all_players
+        self.player_cycle = itertools.cycle(self.all_players)
+
+    def get_current_player(self):
+        if not self.current_player:
+            self.get_next_player()
+        return self.current_player
+
+    def get_next_player(self):
+        self.current_player = self._cycle_next_alive_player()
+
+    def get_alive_player(self):
+        return self.alive_players
+
+    def get_all_other_alive_player(self):
+        return [player for player in self.alive_players if
+                self.current_player != player]
+
+    def _cycle_next_alive_player(self):
+        for player in self.player_cycle:
+            if player in self.alive_players:
+                return player
+            elif not self.alive_players:
+                return None
```

Comment:
This makes your for-loop unnecessary, you're just returning the first player or None if they're alive. You should cycle through and then have this at the end as an error if there are no alive players potentially. However, this should never happen because the last one standing would be considered the winner.

Reply:
The cycle will keep state of the last position so that is untrue.  I have provided an example of what I mean.  Now I realize that I am passing in the cycle object each time but since the cycle is in the self of the class it is the equivalent behavior.

>>> import itertools
>>> a = ['foo', 'bar', 7, 2]
>>> b = itertools.cycle(a)
>>> def cycle_b(cycle_obj):
...  for x in cycle_obj:
...   return x
...
>>> cycle_b(b)
'foo'
>>> cycle_b(b)
'bar'
>>> cycle_b(b)
7
>>> cycle_b(b)
2
>>> cycle_b(b)
'foo'
`

Reply:
This didn't make sense because you're checking whether there are any alive players at every iteration unnecessarily. That should probably be a condition before your for-loop where you raise an exception if there aren't any alive players.